dashboard/src/components/ConfirmDialog/ConfirmDialog.tsx

```diff
@@ -20,35 +21,49 @@ function ConfirmDialog({
   modalIsOpen,
   loading,
   extraElem,
+  headerText,
   confirmationButtonText,
   confirmationText,
   onConfirm,
   closeModal,
   error,
+  size,
 }: IConfirmDialogProps) {
   return (
-    <Modal showModal={modalIsOpen} onModalClose={closeModal}>
-      {error && <Alert theme="danger">An error ocurred: {error.message}</Alert>}
-      {loading === true ? (
-        <div className="confirmation-modal">
-          <span>Loading, please wait</span>
-          <LoadingWrapper loaded={false} />
-        </div>
-      ) : (
-        <div className="confirmation-modal">
-          <span>{confirmationText}</span>
-          {extraElem}
-          <div className="confirmation-modal-buttons">
-            <CdsButton action="outline" type="button" onClick={closeModal}>
-              Cancel
-            </CdsButton>
-            <CdsButton status="danger" type="submit" onClick={onConfirm}>
-              {confirmationButtonText || "Delete"}
-            </CdsButton>
-          </div>
-        </div>
+    <>
+      {modalIsOpen && (
+        <CdsModal size={size || "default"} closable={true} onCloseChange={closeModal}>
+          <CdsModalHeader>{headerText}</CdsModalHeader>
+          {error && <Alert theme="danger">An error ocurred: {error.message}</Alert>}
+          {loading === true ? (
+            <>
+              <CdsModalContent>
+                <span>Loading, please wait</span>
+                <LoadingWrapper loaded={false} />
+              </CdsModalContent>
+            </>
+          ) : (
+            <>
+              <CdsModalContent>
+                <p>{confirmationText}</p>
+                <p>{extraElem}</p>
+              </CdsModalContent>
+              <CdsModalActions>
+                {/* TODO(andresmgot): CdsButton "type" property doesn't work, so we need to use a normal <button>
+                  https://github.com/vmware/clarity/issues/5038
```

Comment:
it'd be good to investigate this...

Reply:
Totally agree, perhaps the best option is to create a very simple scenario in a demo environment and pass it through the internal support channel, as I did with the modal issue